Sources/RInAppMessaging/Router.swift

```diff
@@ -132,46 +135,52 @@ internal class Router: RouterType, ViewListenerObserver {
         }
     }
 
-    private func createViewConstructor(for campaign: Campaign, associatedImageData: Data?) -> (() -> BaseView)? {
-        func getPresenter<T>(type: T.Type) -> T? {
-            guard let presenter = self.dependencyManager.resolve(type: type) else {
-                Logger.debug("Error: \(type) couldn't be resolved")
-                assertionFailure()
+    private func getPresenter(for type: CampaignDisplayType) -> BaseViewPresenterType? {
+        var presenter: BaseViewPresenterType?
+        switch type {
+        case .modal, .full:
+            guard let resolvedPresenter = resolvePresenter(type: FullViewPresenterType.self) else {
                 return nil
             }
-            return presenter
+            presenter = resolvedPresenter
+        case .slide:
+            guard let resolvedPresenter = resolvePresenter(type: SlideUpViewPresenterType.self) else {
+                return nil
+            }
+            presenter = resolvedPresenter
+        default:
+            Logger.debug("Error: Campaign view type is not supported")
         }
+        return presenter
+    }
 
-        switch campaign.data.type {
-        case .modal:
-            guard let presenter = getPresenter(type: FullViewPresenterType.self) else {
-                break
-            }
-            presenter.campaign = campaign
-            if let associatedImageData = associatedImageData {
-                presenter.associatedImage = UIImage(data: associatedImageData)
-            }
-            return { ModalView(presenter: presenter) }
-        case .full:
-            guard let presenter = getPresenter(type: FullViewPresenterType.self) else {
-                break
-            }
+    private func resolvePresenter<T>(type: T.Type) -> T? {
+        guard let presenter = self.dependencyManager.resolve(type: type) else {
+            Logger.debug("Error: \(type) couldn't be resolved")
+            return nil
+        }
+        return presenter
+    }
+
+    private func createViewConstructor(for campaign: Campaign, presenter: BaseViewPresenterType, associatedImageData: Data?) -> (() -> BaseView) {
+        var view: (() -> BaseView)!
+        let type = campaign.data.type
+
+        if type == .modal || type == .full {
```

Comment:
`switch` would be more preferable here. Also, even if it's unrealistic, we need to handle the case of `.invalid` and `.html` types. Currently this function will crash in case of any of those types. (Also in the future we will add html type support)

Reply:
Updated here [57d4646](https://github.com/rakutentech/ios-inappmessaging/pull/275/commits/57d464617318d1c37bbc0e36be3d5a49f81e4b83)